options/validate.go

```diff
@@ -58,6 +59,20 @@ func ValidateFiltering(f *query.Filtering, messageInfo map[string]FilteringOptio
 				return fmt.Errorf("Got invalid literal type for %s, expect %s", fieldTag, fieldInfo.ValueType)
 			}
 
+			if fieldInfo.ValueType == QueryValidate_STRING {
+				if x.Type == query.StringCondition_MATCH {
+					v := x.GetValue()
+					if _, err := regexp.Compile(v); err != nil {
+						return fmt.Errorf("incorrect regex %q in field %q: %w", v, fieldTag, err)
+					}
+					// Check for PostgreSQL POSIX incompatible patterns
```

Comment:
We don't necessarily want to tie our validation to a specific SQL engine if we can avoid it, although it might be fine here.
Is there a reference to how the PostgresQL regex validation is different from what Golang parsing does? This looks like a lot of differences, and having a reference linked here might be more convenient than trying to understand just by looking at all of these conditions.